pallets/author-inherent/src/lib.rs

```diff
@@ -116,6 +121,16 @@ pub mod pallet {
 		pub fn kick_off_authorship_validation(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
 			ensure_none(origin)?;
 
+			// First check that the slot number is valid (greater than the previous highest)
+			let slot = T::SlotBeacon::slot();
+			assert!(
+				slot > HighestSlotSeen::<T>::get(),
+				"Block invalid; Supplied slot number is not high enough"
+			);
+
+			HighestSlotSeen::<T>::put(slot);
```

Comment:
writes should go below the last panic? maybe not necessary but feels right

Reply:
addressed in [f3f9cdd](https://github.com/PureStake/nimbus/pull/41/commits/f3f9cddd07f1b89c40aedcc13305b16c1020b4da)